src/OmniSharp/Settings/FromBodyApplicationModelConvention.cs

```diff
@@ -15,8 +15,8 @@ public void Apply(ApplicationModel application)
                     foreach (var parameter in action.Parameters)
                     {
                         if (parameter.BindingInfo?.BindingSource != null ||
-                            parameter.Attributes.OfType<IBindingSourceMetadata>().Any() ||
-                            ValueProviderResult.CanConvertFromString(parameter.ParameterInfo.ParameterType))
+                            parameter.Attributes.OfType<IBindingSourceMetadata>().Any())
+                            //|| ValueProviderResult.CanConvertFromString(parameter.ParameterInfo.ParameterType))
```

Comment:
the method is needed to not apply `[FromBody]` to string, int or any other string-convertible parameters - only complex types should come from body